dhall-to-yaml/Main.hs

```diff
@@ -31,31 +30,15 @@ data Options = Options
     } deriving (Generic, ParseRecord)
 
 main :: IO ()
-main = handle (do
+main = handle $ do
     GHC.IO.Encoding.setLocaleEncoding GHC.IO.Encoding.utf8
-    Options {..} <- Options.Generic.getRecord "Compile Dhall to JSON"
+    Options{..} <- Options.Generic.getRecord "Compile Dhall to JSON"
 
-    (if Options.Generic.unHelpful explain then Dhall.detailed else id) (do
-        inText <- Data.Text.Lazy.IO.getContents
+    let explaining   = if Options.Generic.unHelpful explain  then Dhall.detailed      else id
+        omittingNull = if Options.Generic.unHelpful omitNull then Dhall.JSON.omitNull else id
 
-        expr <- case Dhall.Parser.exprFromText (Directed "(stdin)" 0 0 0 0) inText of
-            Left  err  -> Control.Exception.throwIO err
-            Right expr -> return expr
-
-        expr' <- Dhall.Import.load expr
-        case Dhall.TypeCheck.typeOf expr' of
-            Left  err -> Control.Exception.throwIO err
-            Right _   -> return ()
-
-        json <- case Dhall.JSON.dhallToJSON expr' of
-            Left err  -> Control.Exception.throwIO err
-            Right json -> return json
-
-        let filteredJSON =
-                if Options.Generic.unHelpful omitNull
-                then Dhall.JSON.omitNull json
-                else json
-        Data.ByteString.putStr (Data.Yaml.encode filteredJSON) ))
+    Data.ByteString.putStr =<< (Data.Yaml.encode . omittingNull <$>) . explaining . (Dhall.JSON.codeToValue "(stdin)")
```

Comment:
I'd prefer to use `do` notation here to split this up over multiple lines.  This is difficult to read

Reply:
Done!